graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ScrollResultES7.java

```diff
@@ -46,13 +44,16 @@ public ScrollResultES7(ElasticsearchClient client,
                            @Assisted List<String> fields,
                            @Assisted int limit) {
         super(client, initialResult, query, fields, limit);
-        checkArgument(initialResult.getScrollId() != null, "Unable to extract scroll id from supplied search result!");
         this.scroll = scroll;
 
     }
 
     @Override
     protected SearchResponse nextSearchResult() throws IOException {
+        if (this.lastSearchResponse.getScrollId() == null) {
+            //with ignore_unavailable=true and no available indices, response does not contain scrollId
+            return null;
```

Comment:
Agreed that it would be cleaner to not return null. But the code that processes the returned value is based on working with null. So are all the tests. We'd have to refactor a bunch more in order to support `Optional`. 
I went ahead and annotated as `Nullable`.
https://github.com/Graylog2/graylog2-server/blob/f04fe4e6dd01f55fcb6ca06d49a05c798109498a/graylog2-server/src/main/java/org/graylog2/indexer/results/ChunkedQueryResult.java#L67-L77